Tests/FoundationEssentialsTests/URLTests.swift

```diff
@@ -340,13 +340,39 @@ final class URLTests : XCTestCase {
 
     #if os(Windows)
     func testURLWindowsDriveLetterPath() throws {
-        let url = URL(filePath: "C:\\test\\path", directoryHint: .notDirectory)
+        var url = URL(filePath: #"C:\test\path"#, directoryHint: .notDirectory)
         // .absoluteString and .path() use the RFC 8089 URL path
         XCTAssertEqual(url.absoluteString, "file:///C:/test/path")
         XCTAssertEqual(url.path(), "/C:/test/path")
         // .path and .fileSystemPath strip the leading slash
         XCTAssertEqual(url.path, "C:/test/path")
         XCTAssertEqual(url.fileSystemPath, "C:/test/path")
+
+        url = URL(filePath: #"C:\"#, directoryHint: .isDirectory)
+        XCTAssertEqual(url.absoluteString, "file:///C:/")
+        XCTAssertEqual(url.path(), "/C:/")
+        XCTAssertEqual(url.path, "C:/")
+        XCTAssertEqual(url.fileSystemPath, "C:/")
+
+        url = URL(filePath: #"C:\\\"#, directoryHint: .isDirectory)
+        XCTAssertEqual(url.absoluteString, "file:///C:///")
+        XCTAssertEqual(url.path(), "/C:///")
+        XCTAssertEqual(url.path, "C:/")
+        XCTAssertEqual(url.fileSystemPath, "C:/")
+
+        url = URL(filePath: #"\C:\"#, directoryHint: .isDirectory)
+        XCTAssertEqual(url.absoluteString, "file:///C:/")
+        XCTAssertEqual(url.path(), "/C:/")
+        XCTAssertEqual(url.path, "C:/")
+        XCTAssertEqual(url.fileSystemPath, "C:/")
+
+        let base = URL(filePath: #"\d:\path\"#, directoryHint: .isDirectory)
```

Comment:
Is Foundation supposed to take this path?

Reply:
I think we should handle filePath inputs of `\d:\` and `d:\` equivalently on Windows since both will be standardized to the same RFC URL path.